### ISSUE TITLE
Element does not trust my friends's keys

### ISSUE THREAD

**Issue by monnier** [vector-im/element-android#2125]
I recently switched from Riot to Element, and while my friends's sessions were marked as trusted in Riot, now they are all marked as untrusted in Element, which means that I cannot send them encrypted messages until I re-do the dance to trust those keys, even tho I already did.

Worse: now I cannot just tell Element to trust the keys, I have to go through an interactive key signing with the other user, which is nigh on impossible with most of my friends since they are rarely connected at the same time as I am (which was the main reason I started using Matrix rather than XMPP).

I'd have expected that all the remote users's keys that were trusted in Riot are still trusted in Element (that information about which correspondant's keys are trusted should presumably be stored on the Matrix server rather than on the client, no?).

I really appreciate the effort to provide strong security, but I think Element would gain a lot by having several levels of security: currently it's either no encryption or encryption with a very painful key distribution scheme.  I can see situations where that distribution scheme makes sense, but for mildly private conversations among friends it's way overkill.  It's getting to the point where Riot/Matrix/Element becomes a recurring joke among my friends.

**Comment by BillCarsonFr** [comment]
Hi @monnier 
>  they are all marked as untrusted in Element, which means that I cannot send them encrypted messages

You can send encrypted message to unverified users. What do you mean exactly?

**Comment by monnier** [comment]
@BillCarsonFr 
Oh, indeed, I did not describe it correctly: what I meant is that my friends cannot decipher the messages I send from Element (whereas they can decipher the messages I send from Riot).

**Comment by BillCarsonFr** [comment]
@monnier  Can you check your security&Privacy settings?
Is this flag on?

<img width="394" alt="image" src="https://user-images.githubusercontent.com/9841565/93769396-036ccd00-fc1b-11ea-9174-d7de8dcfbe9a.png">

**Comment by monnier** [comment]
@BillCarsonFr: Yes, I have this flag enabled.  Because indeed if my friend's sessions were not verified than I wouldn't want them to see the message's content.  But these sessions have been verified already (from Riot)

**Comment by BillCarsonFr** [comment]
So you'd like to be able to mark manually as verified your friend devices?
It's not possible actually, you have to verify the other user once, and then you will trust his trusted devices.

Verification has to be done out of band with the other user for it to be efficient. If you have no interest in verification it's fine to let this flag to off.

**Comment by monnier** [comment]
@BillCarsonFr: So, IIUC, this works fundamentally different in Riot and in Element?

Do I understand it right that in Riot, I had to mark as verified each device individually (and I could do it either interactively or by a manual "override"), whereas now in Element, this verification is "per user" rather than "per user's device" (and it can only be done interactively)?

If so, it sounds like a good change, but the UI does not make it clear at all, since the interactive dance is initiated by clicking on a particular "session" which suggest it will only mark as trusted this particular session rather than all the sessions of that user.
